### PR TITLE
Edit IOException docs in ZipFile.ExtractToDirectory

### LINKED ISSUE #53609: Incorrect documented IOException in ZipFile.ExtractToDirectory
In `ZipFile.ExtractToDirectory` and `ZipFileExtensions.ExtractToDirectory`, the remarks and documentation say that IOException is thrown if the destination directory already exists. This is incorrect, as the method that's eventually called, the internal [ExtractRelativeToDirectory](https://github.com/dotnet/runtime/blob/01b7e73cd378145264a7cb7a09365b41ed42b240/src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileExtensions.ZipArchiveEntry.Extract.cs#L87) method works fine with existing destination folders.

This was first asked in [this SO question](https://stackoverflow.com/questions/67806595/does-zipfile-extracttodirectory-throw-an-exception-when-the-directory-exists/67807550). I also opened an [issue in the dotnett-api-docs repo](https://github.com/dotnet/dotnet-api-docs/issues/6788)

https://github.com/dotnet/runtime/blob/0e557e8fb5b37fbbc57c0a56cae8cfd1ce61b610/src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFile.Extract.cs#L64

### PR DESCRIPTION
**Pull request by steveberdy**
## Summary

Removed docs for `IOException`s getting thrown when the extracted directory already exists, as that is handled elsewhere.

Fixes #53609

### CODE REVIEW

File: src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFile.Extract.cs [hunk 1]
```diff
@@ -28,8 +28,7 @@ public static partial class ZipFile
         /// and file names must be less than 260 characters.</exception>
         /// <exception cref="DirectoryNotFoundException">The path specified by sourceArchive or destinationDirectoryName is invalid,
         /// (for example, it is on an unmapped drive).</exception>
-        /// <exception cref="IOException">The directory specified by destinationDirectoryName already exists.
-        /// -or- An I/O error has occurred. -or- An archive entry?s name is zero-length, contains only whitespace, or contains one or
+        /// <exception cref="IOException">An I/O error has occurred. -or- An archive entry?s name is zero-length, contains only whitespace, or contains one or
```
Comment (jeffhandley) [inline src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFile.Extract.cs:31]:
```suggestion
        /// <exception cref="IOException">An I/O error has occurred. -or- An archive entry's name is zero-length, contains only whitespace, or contains one or
```

File: src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFile.Extract.cs [hunk 2]
```diff
@@ -61,8 +60,7 @@ public static partial class ZipFile
...
```
Comment (jeffhandley) [inline src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFile.Extract.cs:63]:
```suggestion
        /// <exception cref="IOException">An I/O error has occurred. -or- An archive entry's name is zero-length, contains only whitespace, or contains one or
```

File: src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFile.Extract.cs [hunk 3]
```diff
@@ -95,8 +93,7 @@ public static partial class ZipFile
...
```
Comment (jeffhandley) [inline src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFile.Extract.cs:96]:
```suggestion
        /// <exception cref="IOException">An I/O error has occurred. -or- An archive entry's name is zero-length, contains only whitespace, or contains one or
```

File: src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFile.Extract.cs [hunk 4]
```diff
@@ -150,8 +147,7 @@ public static partial class ZipFile
...
```
Comment (jeffhandley) [inline src/libraries/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFile.Extract.cs:150]:
```suggestion
        /// <exception cref="IOException">An I/O error has occurred. -or- An archive entry's name is zero-length, contains only whitespace, or contains one or
```

**Review by jeffhandley — APPROVED** [review]
I am going to commit some suggestions that fix an encoded character issue. Then I'll merge this once the checks run. Thanks!

**Comment by steveberdy** [comment]
Thank you!

**Pull request merged**